/*
 * This file was automatically generated by EvoSuite
 * Thu May 12 14:35:39 GMT 2022
 */

package math;

import org.junit.Test;
import static org.junit.Assert.*;
import static org.evosuite.runtime.EvoAssertions.*;
import math.IntMath;
import org.evosuite.runtime.EvoRunner;
import org.evosuite.runtime.EvoRunnerParameters;
import org.junit.runner.RunWith;

@RunWith(EvoRunner.class) @EvoRunnerParameters(mockJVMNonDeterminism = true, useVFS = true, useVNET = true, resetStaticState = true, separateClassLoader = true, useJEE = true) 
public class IntMath_ESTest extends IntMath_ESTest_scaffolding {

  @Test(timeout = 4000)
  public void test0()  throws Throwable  {
      IntMath intMath0 = new IntMath();
      int int0 = intMath0.mod((-3030), 1864);
      assertEquals(698, int0);
  }

  @Test(timeout = 4000)
  public void test1()  throws Throwable  {
      IntMath intMath0 = new IntMath();
      // Undeclared exception!
      try { 
        intMath0.mod((-1322), 0);
        fail("Expecting exception: ArithmeticException");
      
      } catch(ArithmeticException e) {
         //
         // Modulus 0 must be > 0
         //
         verifyException("math.IntMath", e);
      }
  }

  @Test(timeout = 4000)
  public void test2()  throws Throwable  {
      IntMath intMath0 = new IntMath();
      int int0 = intMath0.mod((-2003), 1);
      assertEquals(0, int0);
  }

  @Test(timeout = 4000)
  public void test3()  throws Throwable  {
      IntMath intMath0 = new IntMath();
      int int0 = intMath0.mod(3745, 4033);
      assertEquals(3745, int0);
  }

  @Test(timeout = 4000)
  public void test4()  throws Throwable  {
      IntMath intMath0 = new IntMath();
      // Undeclared exception!
      try { 
        intMath0.mod((-288), (-288));
        fail("Expecting exception: ArithmeticException");
      
      } catch(ArithmeticException e) {
         //
         // Modulus -288 must be > 0
         //
         verifyException("math.IntMath", e);
      }
  }
}
